src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1141,7 +1146,20 @@ private void DynamoReadyExtensionHandler(ReadyParams readyParams, IEnumerable<IE
             {
                 try
                 {
+                    //check if the extension added a python engine
+                    var pyEngCount = GetEngineList().Count;
                     ext.Ready(readyParams);
+                    var availableEngines = GetEngineList();
+                    if (availableEngines.Count > pyEngCount)
+                    {
+                        //this extension added a python engine, assuming it only added 1 engine
+                        if (PythonEngineExtensionMap == null)
+                        {
+                            PythonEngineExtensionMap = new Dictionary<string, IExtension>();
+                        }
+                        var newEng = availableEngines.Last();
+                        PythonEngineExtensionMap.TryAdd(newEng, ext);
+                    }
```

Comment:
you shouldn't need this if you just hook into the AvailableEngines.CollectionChanged event.